### PR TITLE
Add checkbox field to make a DOI request upon the creation of a new c…

### PR DESCRIPTION
**Pull request by tingtingjh**
…ollection

Fixes: DOI request for a collection without creator, publisher and publicationdate fields; Delete acollection without breaking the DOI admin page

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class CollectionsController < ApplicationController
+  include Sufia::CollectionsControllerBehavior
+
+  def create
+    @collection.apply_depositor_metadata(current_user.user_key)
+    add_members_to_collection unless batch.empty?
+    if @collection.save
+      if params[:request_doi]
+        @collection[:identifier] << "doi:pending"
```
Comment (rotated8) [inline app/controllers/collections_controller.rb]: Please add to a locale file.

File: app/controllers/collections_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        doi_request = DoiRequest.new(asset_type: 'Collection', collection_id: @collection.id)
+        if doi_request.save && @collection.update_attributes({:identifier => @collection[:identifier]})
+          flash[:notice] = "DOI Request has been submitted successfully!"
```
Comment (rotated8) [inline app/controllers/collections_controller.rb]: Please add to a locale file.

File: app/controllers/collections_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        else
+          flash[:error] = "DOI Request error!"
```
Comment (rotated8) [inline app/controllers/collections_controller.rb]: Please add to a locale file.

File: app/views/records/edit_fields/_resource_type.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<%= f.input :resource_type, as: :hidden, input_html: { value: ["Dataset"],class: 'form-control', multiple: true} %>
```
Comment (rotated8) [inline app/views/records/edit_fields/_resource_type.html.erb]: Does this hidden field need to be mulitvalued?

File: app/views/records/edit_fields/_identifier.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<% if f.object.class.multiple? key %>
+  <%= f.input key, as: :multi_value_with_help, input_html: { class: 'form-control' }, required: f.object.required?(key) %>
+<% else %>
+  <%= f.input key, required: f.object.required?(key) %>
+<% end %>
+<%= check_box_tag 'request_doi'%>
+<%= label_tag(:request_doi, "Request a DOI for this Dataset") %>
```
Comment (rotated8) [inline app/views/records/edit_fields/_identifier.html.erb]: Please add to a locale file.

File: app/views/records/show_fields/_identifier.html.erb [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <% record.identifier.each do |id| %>
-  <% if id.match(/^doi/) %>
+  <% if id.match(/^doi/) && id != "doi:pending" %>
```
Comment (rotated8) [inline app/views/records/show_fields/_identifier.html.erb]: This should pull from a locale file.

**Comment by tingtingjh** [comment]
Add static labels and messages to locales.

**Pull request merged**